src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -52,69 +52,427 @@ public class RepoConfiguration {
+        this.isIgnoreGlobListOverriding = repoBuilder.isIgnoreGlobListOverriding;
+        this.isIgnoreCommitListOverriding = repoBuilder.isIgnoreCommitListOverriding;
+        this.isIgnoredAuthorsListOverriding = repoBuilder.isIgnoredAuthorsListOverriding;
+        this.fileSizeLimit = repoBuilder.fileSizeLimit;
+        this.isFileSizeLimitOverriding = repoBuilder.isFileSizeLimitOverriding;
+        this.isIgnoredFileAnalysisSkipped = repoBuilder.isIgnoredFileAnalysisSkipped;
+
+        processAuthor();
+        processBranch();
+        processNames(repoBuilder);
+        processOrganization(repoBuilder);
     }
 
-    public RepoConfiguration(RepoLocation location, String branch) {
-        this(location, branch, DEFAULT_EXTRA_OUTPUT_FOLDER_NAME);
+    /**
+     * Processes the author configuration of the repository.
+     */
+    private void processAuthor() {
+        this.authorConfig = new AuthorConfiguration(location, branch);
     }
 
-    public RepoConfiguration(RepoLocation location, String branch, String extraOutputFolderName) {
-        this(location, branch, Collections.emptyList(), Collections.emptyList(),
-                RepoConfiguration.DEFAULT_FILE_SIZE_LIMIT, false, false, Collections.emptyList(), false, false, false,
-                false, false, false, false, Collections.emptyList(), false, extraOutputFolderName);
+    /**
+     * Processes the branch of the repository.
+     */
+    private void processBranch() {
+        this.branch = this.location.isEmpty() ? DEFAULT_BRANCH : this.branch;
     }
 
-    public RepoConfiguration(RepoLocation location, String branch, List<FileType> formats, List<String> ignoreGlobList,
-            long fileSizeLimit, boolean isStandaloneConfigIgnored, boolean isFileSizeLimitIgnored,
-            List<CommitHash> ignoreCommitList, boolean isFormatsOverriding, boolean isIgnoreGlobListOverriding,
-            boolean isIgnoreCommitListOverriding, boolean isFileSizeLimitOverriding, boolean isShallowCloningPerformed,
-            boolean isFindingPreviousAuthorsPerformed, boolean isIgnoredFileAnalysisSkipped,
-            List<String> ignoredAuthorsList, boolean isIgnoredAuthorsListOverriding) {
-        this(location, branch, formats, ignoreGlobList, fileSizeLimit, isStandaloneConfigIgnored,
-                isFileSizeLimitIgnored, ignoreCommitList, isFormatsOverriding, isIgnoreGlobListOverriding,
-                isIgnoreCommitListOverriding, isFileSizeLimitOverriding, isShallowCloningPerformed,
-                isFindingPreviousAuthorsPerformed, isIgnoredFileAnalysisSkipped, ignoredAuthorsList,
-                isIgnoredAuthorsListOverriding, DEFAULT_EXTRA_OUTPUT_FOLDER_NAME);
+    /**
+     * Processes the relevant names of the repository configs.
+     */
+    private void processNames(RepoBuilder repoBuilder) {
+        String repoName = this.location.getRepoName();
+        this.displayName = repoBuilder.displayName == null
```

Comment:
would an `Optional` be a good idea for these? we probably should be using more `Optional`s in RepoSense to enforce a check, but that's a separate discussion...

Reply:
I agree, `Optional` would eliminate the `== null` checks, making the code a lot more understandable and readable! Will look into using`Optional` here!

Reply:
Should we look into creating an issue for these separately?

Reply:
@MarcusTXK I agree, most of the issues we had discussed above are rather complicated (parameter verification, duplication of `RepoConfiguration`, etc.) and seem to warrant the creation of separate issues.

Reply:
Yeah I think we can focus on improving the `RepoConfiguration` incrementally, as @gok99 has brought up alot of good points.